custom/icds_reports/reports/disha.py

```diff
@@ -122,5 +123,19 @@ def build_dumps_for_month(month, rebuild=False):
             logger.info("Skipping, export is already generated for state {}".format(state_name))
         else:
             logger.info("Generating for state {}".format(state_name))
-            dump.build_export_json()
+            MAX_RETRY_COUNT = 5
+            retry_count = 0
+            while True:
+                try:
+                    dump.build_export_json()
+                except Exception as e:
```

Comment:
nit: i think this better reads as
```
try:
    dump.build_export_json()
except Exception as e:
    if is_pg_cancelled_query_exception(e) and retry_count < MAX_RETRY_COUNT:
        retry_count += 1
        logger.info("Postgres cancelled the DISHA query. Retry count: {}".format(str(retry_count)))
        continue
    else:
        raise e
else:
    break
```

or if the following would work as expected

```
try:
    dump.build_export_json()
    break
except Exception as e:
    typ, exc, tb = sys.exc_info()
    if is_pg_cancelled_query_exception(e) and retry_count < MAX_RETRY_COUNT:
        retry_count += 1
        logger.info("Postgres cancelled the DISHA query. Retry count: {}".format(str(retry_count)))
        continue
    six.reraise(typ, exc, tb)
```